### PR TITLE
Web console: adding format notice for CSV and TSV

### PR DESCRIPTION
**Pull request by vogievetsky**
Adds a warning that the CSV or TSV sample contains newlines in values which is allowed in the format but Druid will not accept it in the expected way. Adding this because I got really tripped up by this myself.

image

Try it with:

```csv
"a","b","c","d","e"
"hello
world",2,3,4,5
```

Feedback on copy very welcome.

Also changed the links from http://ndjson.org/ to https://jsonlines.org (due to https://github.com/ndjson/ndjson-spec/issues/35#issuecomment-1285673417)

### CODE REVIEW

File: web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx [hunk 1]
```diff
@@ -2135,33 +2136,53 @@ export function updateIngestionType(
...
-      <>
-        This data looks like multi-line formatted JSON object. For Druid to parse a text file it
-        must have one row per event. Consider reformatting your data as{' '}
-        <ExternalLink href="http://ndjson.org/">newline delimited JSON</ExternalLink>.
-      </>
-    );
+    if (oneOf(firstLine, '[', '[]')) {
+      return (
+        <>
+          This data looks like a multi-line JSON array. For Druid to parse a text file it must have
```
Comment (317brian) [inline web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx:2164]:
```suggestion
          This data looks like a multi-line JSON array. For Druid to parse a text file, it must have
```

File: web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx [hunk 2]
```diff
@@ -2135,33 +2136,53 @@ export function updateIngestionType(
...
+  const text = sampleLines.join('\n');
+  if (oneOf(format.type, 'csv', 'tsv')) {
+    const valueWithNewline = findValueWithNewline(
+      format.type === 'csv' ? csvParseRows(text) : tsvParseRows(text),
     );
+    if (valueWithNewline) {
+      const formatLabel = format.type.toUpperCase();
+      return (
+        <>
+          {`This ${formatLabel} data has values that contain new lines. Druid requires ${formatLabel} files to have one event per line and thus ${formatLabel} values can not contain new lines. Consider encoding new lines in the values of your ${formatLabel} with some special delimiter.`}
```
Comment (317brian) [inline web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx:2182]:
```suggestion
          {`This ${formatLabel} data has values that contain new lines. Druid requires ${formatLabel} files to have one event per line, so ${formatLabel} values can not contain new lines. Consider encoding new lines in the values of your ${formatLabel} with some special delimiter.`}
```

File: web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx [hunk 3]
```diff
@@ -2135,33 +2136,53 @@ export function updateIngestionType(
...
+  if (!sampleLines.length) return;
 
-  const firstData: string = findMap(sampleData.data, l => l.input?.raw);
-  if (firstData) return;
+  const firstLine = sampleLines[0];
+  if (!isStreaming) {
+    if (firstLine === '{') {
+      return (
+        <>
+          This data looks like multi-line formatted JSON object. For Druid to parse a text file it
```
Comment (317brian) [inline web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx:2154]:
```suggestion
          This data looks like multi-line formatted JSON object. For Druid to parse a text file, it
```

**Review by 317brian — CHANGES_REQUESTED** [review]

File: web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx (hunk 3 above)
Comment (ektravel) [inline web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx:2154]:
```suggestion
          This data looks like a multi-line formatted JSON object. For Druid to parse a text file it
```

File: web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx (hunk 2 above)
Comment (ektravel) [inline web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx:2182]:
```suggestion
          {`This ${formatLabel} data has values that contain new lines. Druid requires ${formatLabel} files to have one event per line, so ${formatLabel} values cannot contain new lines. Consider encoding new lines in the values of your ${formatLabel} with some special delimiter.`}
```

**Review by ektravel — COMMENTED** [review]
Reviewed the copy.

**Review by clintropolis — APPROVED** [review]

**Pull request merged**